docs/specs/user-profile/design.md

```diff
@@ -0,0 +1,329 @@
+# User profile
+
+User profile consists of standard attributes and custom attributes.
+
+## Standard attributes
+
+Standard attributes is a subset of [OIDC standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims).
```

Comment:
If we support all the name attributes, namely given_name, family_name, middle_name, name and nickname,
there are too many name fields for the end-user to fill in.
For example, Google supports given_name, family_name and name only. https://developers.google.com/identity/protocols/oauth2/openid-connect

The profile attribute is just an URL.
Google supports it. https://developers.google.com/identity/protocols/oauth2/openid-connect
Facebook does NOT support it. https://developers.facebook.com/docs/graph-api/reference/user/#default-public-profile-fields

The picture attribute is just an URL.
Google supports it. https://developers.google.com/identity/protocols/oauth2/openid-connect
Facebook supports it. https://developers.facebook.com/docs/graph-api/reference/user/#default-public-profile-fields
I tend not to support allowing the end-user to upload their profile picture.
That would require us to maintain an object storage just for that purpose.
The simpliest way to have minimal support is support picture attribute with our automatic population.

The website attribute is just an URL.
Google does NOT support it. https://developers.google.com/identity/protocols/oauth2/openid-connect
Facebook does NOT support it. https://developers.facebook.com/docs/graph-api/reference/user/#default-public-profile-fields

The gender attribute is tricky.
On Facebook you can set it to values other than male or female.
Different application have different requirement on the possible values of this attribute.
It takes a lot of effort to support the gender attribute in a socially inclusive way.

The address attribute is hard to get it right.
Different country or region has different addressing system.
On Amazon, the end-user first choose the country or region, and then the address fields are changed.
It takes a lot of effort to get it right.
Amazon needs to get it right because it needs their customers addresses to ship goods, but we do not.
It is better to leave this to the application.

According to the OIDC spec, the updated_at is the time end-user information changes.
Since we do not have user profile now, we do not have updated_at.
Of course we can add it.

Reply:
So I don’t see any reason beyond it is a bit more difficult to implement…? 

Use case wise I think the minimal is we should add support for profile pic, which is pretty commonly expected when there is a user profile between multiple apps share same login